src/common/dav/wal_tx.c

```diff
@@ -259,11 +272,17 @@ dav_wal_tx_set_bits(void *hdl, void *addr, uint32_t pos, uint16_t num_bits)
 	struct dav_obj		*dav_hdl = (struct dav_obj *)hdl;
 	struct dav_tx		*tx = utx2wtx(dav_hdl->do_utx);
 	struct wal_action	*wa_redo;
+	int                      rc;
 
 	D_ASSERT(hdl != NULL);
 	if (addr == NULL)
 		return -DER_INVAL;
 
+	rc = umem_cache_touch(dav_hdl->do_store, dav_hdl->do_utx->utx_id,
+			      mdblob_addr2offset(tx->wt_dav_hdl, addr), sizeof(uint64_t));
```

Comment:
interesting, collocated metadata can reduce the write amplification, but it could potentially increase the chance of TX restart while checkpointing.